notebooks/llm-chatbot/ov_rerank_model.py

```diff
@@ -0,0 +1,164 @@
+import numpy as np
+from pathlib import Path
+from optimum.intel.openvino import OVModelForSequenceClassification
+from langchain.callbacks.manager import Callbacks
+from langchain_core.documents import Document
+from langchain.retrievers.document_compressors.base import BaseDocumentCompressor
+from typing import Optional, Dict, Any, Sequence
+from langchain.pydantic_v1 import Field
+import json
+from pathlib import Path
+from tokenizers import AddedToken, Tokenizer
+import numpy as np
+import collections
+
+
+class RerankRequest:
+
+    def __init__(self, query=None, passages=None):
+        self.query = query
+        self.passages = passages if passages is not None else []
+
+
+class OVRanker(BaseDocumentCompressor):
+    """
+    OpenVINO BGE rerank models.
+
+    """
+
+    ov_model: Any
+    """OpenVINO model object."""
+    tokenizer: Any
+    """Tokenizer for embedding model."""
+    model_dir: str
+    """Path to store models."""
+    model_kwargs: Dict[str, Any]
+    """Keyword arguments passed to the model."""
+    top_n: int = 4
+    """return Top n texts."""
+
+    def __init__(self, **kwargs: Any):
+        super().__init__(**kwargs)
+        self.tokenizer = self._get_tokenizer()
+        self.ov_model = OVModelForSequenceClassification.from_pretrained(
+            self.model_dir, **self.model_kwargs)
+
+    def _load_vocab(self, vocab_file):
+
+        vocab = collections.OrderedDict()
+        with open(vocab_file, "r", encoding="utf-8") as reader:
+            tokens = reader.readlines()
+        for index, token in enumerate(tokens):
+            token = token.rstrip("\n")
+            vocab[token] = index
+        return vocab
+
+    def _get_tokenizer(self, max_length=512):
+
+        config_path = Path(self.model_dir) / "config.json"
+        if not config_path.exists():
+          raise FileNotFoundError(f"config.json missing in {self.model_dir}")
+
+        tokenizer_path = Path(self.model_dir) / "tokenizer.json"
+        if not tokenizer_path.exists():
+          raise FileNotFoundError(
+              f"tokenizer.json missingin  {self.model_dir}")
+
+        tokenizer_config_path = Path(self.model_dir) / "tokenizer_config.json"
+        if not tokenizer_config_path.exists():
+          raise FileNotFoundError(
+              f"tokenizer_config.json missing in  {Path(self.model_dir)}")
+
+        tokens_map_path = Path(self.model_dir) / "special_tokens_map.json"
+        if not tokens_map_path.exists():
+          raise FileNotFoundError(
+              f"special_tokens_map.json missing in  {Path(self.model_dir)}")
+
+        config = json.load(open(str(config_path)))
```

Comment:
I think better to use
```suggestion
       with config_path.open() as config_file:
        config = json.load(config_file)
```
in this case,python will handle issues related open/close file on own side (while in current used code not)